roles/ci_gen_kustomize_values/tasks/edpm_core_asserts.yml

```diff
@@ -25,3 +25,5 @@
       - cifmw_ci_gen_kustomize_values_ssh_private_key != ''
       - cifmw_ci_gen_kustomize_values_ssh_public_key is defined
       - cifmw_ci_gen_kustomize_values_ssh_public_key != ''
+      - cifmw_ci_gen_kustomize_values_ctlplane_interface is defined
```

Comment:
as I see things, the way we have to ensure we're having the needed parameters won't scale.

Wondering if we should:
- keep the check in the ci-framework (and find a way to make it scalable)
- drop it completely and just let the template fail (but that won't provide any nice message showing what's missing)
- add a new entry in the "stages" defined in the automation file, listing mandatory parameters for the template (maybe without the `cifmw_ci_gen_kustomize_` prefix to make it "generic" ?)

In any cases, we must have that discussion - the sooner the better

cc @abays @pablintino @psathyan @rebtoor @lewisdenny 

Guess we need a proper sync/call about that, an design a solution.

Reply:
> as I see things, the way we have to ensure we're having the needed parameters won't scale.
> 
> Wondering if we should:
> 
> * keep the check in the ci-framework (and find a way to make it scalable)
> * drop it completely and just let the template fail (but that won't provide any nice message showing what's missing)
> * add a new entry in the "stages" defined in the automation file, listing mandatory parameters for the template (maybe without the `cifmw_ci_gen_kustomize_` prefix to make it "generic" ?)
> 
> In any cases, we must have that discussion - the sooner the better
> 
> cc @abays @pablintino @psathyan @rebtoor @lewisdenny
> 
> Guess we need a proper sync/call about that, an design a solution.

@cjeanner 
Moved cifmw_ci_gen_kustomize_values_ctlplane_interface validation only when cifmw_baremetal_hosts is deined.

Reply:
Is this for "any" kind of host, or would this be only for computes?

Asking, because in such a case, that block should go in https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/ci_gen_kustomize_values/tasks/edpm_values.yml

in any cases... we'll have to find a better way.

Reply:
@cjeanner edpm-values only for computes. this link is common and not specific to baremetal computes.

Reply:
OK, so:
- we'll keep the core validations only, since they apply to any cases (we know for now)
- I'll do a follow-up to clean non-core validations
- I'll try to make a doc about those validations and why we probably won't add anything new (core vs specific)
- README will also show that situation: template will fail on its own if you don't provide the expected parameters

That's more sustainable than the current content, and we can't really rely on the `architecture` automation for those validations: it doesn't have  a clue about "what a parameter is", and it should stay as is.